content/en/docs/tutorials/configuration/configure-java-microservice/configure-java-microservice.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "Externalizing config using MicroProfile, ConfigMaps and Secrets"
+content_template: templates/tutorial
+weight: 10
+---
+
+{{% capture overview %}}
+
+In this tutorial you will learn how and why to externalize your microservice’s configuration.  Specifically, you will learn how to use Kubernetes ConfigMaps and Secrets to set environment variables and then consume them using MicroProfile Config.
```

Comment:
Hello @mbroz2 .
Thanks for taking the time to write a new tutorial.
I read through the tutorial content. I tried to run the tutorial, but some of the commands timed out (minikube, JAVA_HOME errors).
Generally, I like the idea of demonstrating how to configure and run Java microservices in a Kubernetes cluster, however, could the tutorial run without MicroProfile and Open Liberty?
I assume this is the goal of the tutorial:
`This demonstrates how communication can be achieved between two microservices in separate pod's inside a Kubernetes cluster`
Could readers download any code samples and run their microservices without Open Liberty?

Reply:
Also, the structure of the docs content pages was updated after you created this PR.
Have a look at another tutorial page to see the new page format or
read the contribute documentation at:
https://kubernetes.io/docs/contribute/style/page-content-types#tutorials.

Reply:
👋 Hey @mbroz2 just checking in to see if you had time to review the above feedback. This PR was been open for awhile and I would like to help it move forward if possible. Thanks!

Reply:
Thank you for taking the time to go through the tutorial and providing the feedback.

The reason for the JAVA_HOME issue seems to have been due to Katacoda changing the location of Java; this has been now resolved.  The reason for the commands timing out were likely because the Katacoda portion of the tutorial is currently on a personal/free-tier account, which I believe provides no guarantee of resources and the more users of Katacoda (like during peak hours), the worse the performance for those accounts.  This should be resolved when the tutorial is moved from the personal account to the Kubernetes account (I believe `kubernetes-bootcamp`) used for the other Katacoda tutorials.  As a side note, I've been trying to reproduce the timeouts, but have been unable to, so it's also possible the issue was due to JAVA_HOME not getting set correctly.
Everything in the Katacoda tutorial should be resolved, but if anything looks amiss, please let me know.

The overall goal of the tutorial is to teach how to leverage Kubernetes Secrets & ConfigMaps to externalize configuration of a Java microservices deployed in a Kubernetes environment.  The `This demonstrates how communication can be achieved between two microservices in separate pod's inside a Kubernetes cluster` was just the scenario that was chosen as the teaching mechanism, but just about any scenario could have been used where a configuration would have been necessary or helpful.
Regarding whether this could be done without MicroProfile, the answer is technically yes, but you probably shouldn't, and if this tutorial took that approach then it would be 'my personal' (and likely not portable, and possible flawed & unreliable) approach on how to do so, rather then an industry standard approach to doing so (MicroProfile is an open-source Enterprise Java standard [headed by the Eclipse Foundation](https://projects.eclipse.org/projects/technology.microprofile), with [much of the Java Enterprise industry closely involved in its development](https://microprofile.io/)).
However, although MicroProfile provides various specs, it does not provide a reference implementation, but one is necessary to run a MP application (similarly how one needs a Java EE complaint server to run an Enterprise Java app).  There are many [implementors of MicroProfile](https://wiki.eclipse.org/MicroProfile/Implementation), and any one of them could be used in the tutorial as the Java runtime.  Open Liberty is one of those implementors, an open-source, container optimized implementation which implements all and remains current with the MicroProfile specs.

Regarding the new docs structure, I've rebased on master to pull in the changes and updated the PR/tutorial.

Reply:
Thanks for the explanation @mbroz2. To recap:
- The tutorial needs a MicroProfile implementation which in this case is provided by IBM's Open Liberty (github.com/OpenLiberty) app server runtime.
- The tutorial needs a Katacoda account.
- It looks like the HTML may need further updating (remove/replace Roboto fonts). See PR #23762.
- How does this tutorial fit in with the Content guide (third party content)? See issue #20232.

Reply:
yes, any MP/JEE app needs a compatible runtime.  Similarly to how [Deploy an App Interactive Tutorial](https://kubernetes.io/docs/tutorials/kubernetes-basics/deploy-app/deploy-intro/) needs a NodeJS server to run its JS app.

yes, similarly to how the other [k.io Interactive Tutorials ](https://kubernetes.io/docs/tutorials/kubernetes-basics/deploy-app/deploy-interactive/) need Katacoda

nice catch, fixed.

None of the content is Open Liberty specific (that's the large point of MicroProfile, to prevent vendor lock-in in the Enterprise Java community/ecosystem).  I'm happy to remove the reference to Open Liberty in the markdown (the interactive tutorial itself makes no reference to Open Liberty).  It was added there for transparency, to let the user know which implementation of MicroProfile was being used, similarly to how [Deploy an App Interactive Tutorial](https://kubernetes.io/docs/tutorials/kubernetes-basics/deploy-app/deploy-intro/) mentions NodeJS.  
So, I think the contents of this tutorial fits within the content guide (third party content), but I could have missed something as it is a lengthy issue/discussion.  For completeness, I'll also mention this is where/how we were we asked to proceed with this content when it was discussed on the SIG-Docs call, but that was before #20232, so things may have changed since.

Reply:
I did just spot that #20232 states `Docs can link to third-party open source software (OSS) outside the Kubernetes project if it’s necessary for Kubernetes to function.` so I'm going to remove any external links (Open Liberty & MicroProfile) as they are not `necessary for Kubernetes to function`

Reply:
Updated the content so that only internal k.io links are present (the Interactive Tutorial does not have any external links).  Added a sentence to clarify that any MP compatible runtime can be used by the developer.